src/solvers/dgsem_tree/dg_2d_parabolic.jl

```diff
@@ -0,0 +1,652 @@
+# This file collects all methods that have been updated to work with parabolic systems of equations
+#
+# assumptions: parabolic terms are of the form div(f(u, grad(u))) and
+# will be discretized first order form as follows:
+#               1. compute grad(u)
+#               2. compute f(u, grad(u))
+#               3. compute div(u)
+# boundary conditions will be applied to both grad(u) and div(u).
+function rhs_parabolic!(du, u, t, mesh::TreeMesh{2}, equations_parabolic::AbstractEquationsParabolic,
+                        initial_condition, boundary_conditions, source_terms,
+                        dg::DG, dg_parabolic, cache, cache_parabolic)
+
+  # Reset du
+  @trixi_timeit timer() "reset ∂u/∂t" reset_du!(du, dg, cache)
+
+
+  @unpack u_transformed, u_grad, viscous_flux = cache_parabolic
+  @trixi_timeit timer() "transform variables" transform_variables!(u_transformed, u,
+                                                                   mesh, equations_parabolic,
+                                                                   dg, dg_parabolic,
+                                                                   cache, cache_parabolic)
+
+  @trixi_timeit timer() "calculate gradient" calc_gradient!(u_grad, u_transformed, t, mesh,
+                                                            equations_parabolic,
+                                                            boundary_conditions, dg,
+                                                            cache, cache_parabolic)
+
+  @trixi_timeit timer() "calculate viscous fluxes" calc_viscous_fluxes!(viscous_flux, u_grad, u_transformed,
```

Comment:
I don't think it's strictly necessary for `DGSEM` solvers. It's necessary for `DGMulti` since we perform interpolation/projection in between computing the viscous flux, and since we need to apply `mul!` to the arrays holding the viscous flux. 

For `DGSEM`, since the boundary nodes are included in the volume nodes, one could compute the viscous interface fluxes on the fly. However, I think it's still a good idea to introduce temporary storage because it may make extending to mortar interfaces easier. Thoughts?

Reply:
Thanks for the clarification. Do you see a structural reason why we would need temporary storage for the mortars? I mean, we did not need it for the hyperbolic fluxes, so what do you have in mind that might be different here?

Reply:
Sure - I'm not 100% sure how the mortars are treated, so please take this with a grain of salt. 

My impressions is that mortar fluxes are computed via something like `flux(u_mortar)`, where `u_mortar` is interpolated from face points to mortar points. Storing the viscous flux as a separate array allows fluxes to be computed via `flux_mortar`, where we just interpolate the stored `viscous_flux`.

Reply:
However, the reason we implemented a temporary storage is because we were copying `DGMulti` at the time of implementation. If it makes more sense to compute fluxes on the fly for `TreeMesh` and `DGSEM`, I am supportive of that.

Reply:
Hm. But wouldn't this then have to hold for the inviscid fluxes as well? Maybe I'm not seeing it. But we could just add it to our issue to rule all issues and leave it as an optimization for later to make the parabolic part more consistent with the hyperbolic part.

Reply:
Ah, sorry. I didn't see you second comment in time. If there is an easy way to remove the temporary storage, I suggest to give it a try - if not, record it for later.

Reply:
Opened an issue in https://github.com/trixi-framework/Trixi.jl/issues/1198.